unstructured/embed/bedrock.py

```diff
@@ -0,0 +1,72 @@
+import types
+from typing import List
+
+import numpy as np
+import boto3
+
+from unstructured.documents.elements import (
+    Element,
+)
+from unstructured.embed.interfaces import BaseEmbeddingEncoder
+from unstructured.ingest.error import EmbeddingEncoderConnectionError
+from unstructured.utils import requires_dependencies
+from langchain.embeddings import BedrockEmbeddings
+
+class BedrockEmbeddingEncoder(BaseEmbeddingEncoder):
+    def __init__(self, aws_access_key_id: str, aws_secret_access_key: str, region_name: str = "us-west-2"):
+        self.aws_access_key_id = aws_access_key_id
+        self.aws_secret_access_key = aws_secret_access_key
+        self.region_name = region_name
+        self.initialize()
+
+    def initialize(self):
+        self.bedrock_client = self.get_bedrock_client()
+
+    def num_of_dimensions(self):
+        return np.shape(self.examplary_embedding)
+
+    def is_unit_vector(self):
+        return np.isclose(np.linalg.norm(self.examplary_embedding), 1.0)
+
+    def embed_query(self, query):
+        return np.array(self.bedrock_client.embed_query(query))
+
+    def embed_documents(self, elements: List[Element]) -> List[Element]:
+        embeddings = [np.array(self.bedrock_client.embed_query(str(e))) for e in elements]
+        elements_with_embeddings = self._add_embeddings_to_elements(elements, embeddings)
+        return elements_with_embeddings
+
+    def _add_embeddings_to_elements(self, elements, embeddings) -> List[Element]:
+        assert len(elements) == len(embeddings)
+        elements_w_embedding = []
+
+        for i, element in enumerate(elements):
+            original_method = element.to_dict
+
+            def new_to_dict(self):
+                d = original_method()
+                d["embeddings"] = self.embeddings
```

Comment:
yeah; given this operation here I mean leaning even more toward moving this function to be part of element (here `self` refers to an element... NOT this class itself